### PR TITLE
[fp8] add fp8 comm for low level zero

### LINKED ISSUE #5960: [fp8] support low level zero

### PR DESCRIPTION
**Pull request by ver217**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

closed #5960

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

add fp8 comm for low level zero.

Tested on BERT finetuning.

| Plugin         | Accuracy | F1-score | GPU number |
| -------------- | -------- | -------- | -------- |
| torch_ddp      | 84.4%    | 88.6%    |    2     |
| low level zero-1 + fp8 | 84.3% | 88.5% | 2 |
| low level zero-2 + fp8 | 84.8% | 88.8% | 2 |

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Review by BurkeHulk — APPROVED** [review]

File: colossalai/zero/low_level/bookkeeping/tensor_bucket.py [hunk 1]
```diff
@@ -61,11 +63,14 @@ def unflatten_and_copy(self, flat_tensor):
...
+    def all_gather(self, group=None, fp8_communication: bool = False):
         flat = self.flatten()
-        buffers = [torch.empty_like(flat) for _ in range(dist.get_world_size(group))]
-        dist.all_gather(buffers, flat, group=group)
-        unflat_buffers = [self.unflatten(buffer) for buffer in buffers]
+        buffer = torch.empty(flat.numel() * dist.get_world_size(group), device=flat.device, dtype=flat.dtype)
+        if fp8_communication:
+            all_gather_into_tensor_flat_fp8(buffer, flat, output_shape=buffer.shape, group=group)
+        else:
+            dist.all_gather_into_tensor(buffer, flat, group=group)
```
Comment (GuangyaoZhang) [inline colossalai/zero/low_level/bookkeeping/tensor_bucket.py:72]: Revert this change
